lib/ansible/modules/iptables.py

```diff
@@ -383,6 +390,28 @@
     type: bool
     default: false
     version_added: "2.13"
+  in_data:
+    description:
+      - Data structure created and returned by this module to be used as a memory
+        store internally or to archive and use as a base.
+      - This data can be found in the output.data registered data.
+    type: list
+    elements: dict
+    default: [{}]
+    version_added: "3.0"
+  picker_definitions:
+    description:
+      - Data structure to create rules based on the same structure being used to
+        define rules from tasks and roles.
+      - Data needs to be passed in for picker_includes to consume.
+    type: dict
+    version_added: "3.0"
+  picker_includes:
+    description:
+      - A dictionary of rule definition names and whether they should be enabled
+        or disabled.
+    type: dict
+    version_added: "3.0"
```

Comment:
in any case the in_data/picker seems to be more appropriate of creating a new module that handles the data in these structures vs the current module which is specific to per rule invocation.

Reply:
I figured having the feature here was appropriate for a few reasons:
- you have already done most of the options for me and I’d just be copying 90% of the code (literally everything except for the stuff that runs/test commands) into a new module.
- there are options from iptables-extensions that aren’t implemented here - I’d like to see that eventually done but don’t really want to do it twice.
- I obviously want the order to match what iptables thinks it should have and figure this would benefit here too.
- Test coverate seems to be missing a few parts that I’d eventually like to look at.

I agree with changing do to persist (didn’t really like saying ‘do’ anyway) and I can/should do ‘add’ vs ‘save’ functionality based on other data that’s provided vs having it as an option anyway. The reason for wanting the ‘do’ in the first place is to not break current functionality. But let me know if you’d like me to take the code I need and create something new - I’m sure I can make licensing work since the end goal is to publish it anyway.

Reply:
Maybe it’s worth pointing out that xtables-multi implements iptables, iptables-save, iptables-restore, and ebtables (other commands I’m forgetting?) so there’s kind of precedence for having multiple functionality under a single umbrella here. Not arguing that other projects should continue with others’ bad practices. But that the netfilter people seem to have found this useful, so maybe it’s also useful here.